gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -592,6 +604,39 @@ public boolean handle(final RoutingContext ctx) {
         }
     }
 
+    /**
+     * Handles hook search requests based on the 'destination' property.
+     * Searches in both routes and listeners.
+     *
+     * @param queryParam the RoutingContext of the request
+     */
+    public void handleHookSearch(String queryParam,HttpServerResponse response) {
```

Comment:
Why do you search for routes and listeners? It is not very efficient when you make a `GET` to `/playground/server/hooks/v1/registrations/listeners?q=foobar` and then search for matching routes.

Reply:
Ok, I will create a separate Endpoints for Listeners and Routes